kastle/connection.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+from typing import Optional, Dict
+from functools import cached_property
+
+from . import errors
+
+CRLN = b"\r\n"
+
+
+class Request:
+    __slots__ = ("verb", "target", "version", "_headers", "_body")
+
+    def __init__(self, verb: bytes, target: bytes, version: bytes) -> None:
+        self.verb = verb
+        self.target = target
+        self.version = version
+
+        self._headers = {}
+        self._body = b''
+
+    @property
+    def headers(self) -> Dict[bytes, bytes]:
+        if type(self._headers) is not dict:
+            temp = {}
+
+            for header in self._headers:
+                name, value = header.split(b':', 1)
+                temp[name] = value
+
+            self._headers = temp
+
+        return self._headers
+
+    @property
+    def body(self) -> bytes:
+        return self._body
+
+
+class Connection:
+    __slots__ = ("server", "reader", "writer")
+
+    def __init__(self,
+                 s,
+                 r: asyncio.StreamReader,
+                 w: asyncio.StreamWriter) -> None:
+
+        self.server = s
+        self.reader = r
+        self.writer = w
+
+    async def close(self) -> None:
+        self.writer.close()
+
+    def send(self, data: bytes) -> None:
+        self.writer.write(data)
+        return self.writer.drain()
+
+    async def parse(self) -> None:
+        firstline = await self.reader.readline()
+
+        try:
+            verb, target, version = firstline.split(b' ')
```

Comment:
```suggestion
            method, target, version = firstline.split(b' ')
```
Since this is a HTTP method.